rmw_connextdds_common/src/ndds/dds_api_ndds.cpp

```diff
@@ -197,25 +200,119 @@ rmw_connextdds_initialize_participant_qos_impl(
   dp_qos->user_object.topic_user_object.size = sizeof(void *);
   dp_qos->user_object.content_filtered_topic_user_object.size = sizeof(void *);
 #endif /* RMW_CONNEXT_SHARE_DDS_ENTITIES_WITH_CPP */
+  if (!ctx->domain_tag) {
+    const auto pid = rcutils_get_pid();
+    ctx->domain_tag = DDS_String_alloc(18 + log10(pid));
+    sprintf(ctx->domain_tag, "ros_discovery_off_%d", pid);
+  }
 
   switch (ctx->participant_qos_override_policy) {
     case rmw_context_impl_t::participant_qos_override_policy_t::All:
     case rmw_context_impl_t::participant_qos_override_policy_t::Basic:
       {
         // Parse and apply QoS parameters derived from ROS 2 configuration options.
+        // Reference link for properties:
+        // https://community.rti.com/static/documentation/connext-dds/6.1.1/doc/manuals/connext_dds_professional/properties_reference/index.html
+        if (ctx->discovery_options) {
+          const auto range = ctx->discovery_options->automatic_discovery_range;
+          switch (range) {
+            case RMW_AUTOMATIC_DISCOVERY_RANGE_SUBNET:
+              /* No action needed. This is the default discovery behavior for DDS */
+              break;
+            default:
+              RMW_CONNEXT_LOG_WARNING_A(
+                "Unknown value provided for automatic discovery range: %i",
+                ctx->discovery_options->automatic_discovery_range);
+              /* Fall back to the default behavior */
+              [[clang::fallthrough]];
+              /* [[fallthrough]]; // Uncomment this when migrating to C++17 */
```

Comment:
Same comment as on the other pr, we have access to C++17, this should either be removed or addressed I think.

Reply:
43e6382